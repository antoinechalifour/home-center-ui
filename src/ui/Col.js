import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Col = ({ children, ...props }) => {
  const kids = React.Children.map(children, child => {
    if (child.props.flex) {
      const flex = typeof child.props.flex === 'number' ? child.props.flex : 1
      const style = child.props.style ? { ...child.props.style } : {}

      style.flex = flex

      return React.cloneElement(child, { style })
    } else {
      return child
    }
  })

  return <Wrapper {...props}>{kids}</Wrapper>
}

Col.propTypes = {
  align: PropTypes.string.isRequired,
  justify: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.element).isRequired
}

Col.defaultProps = {
  align: 'flex-start',
  justify: 'flex-start'
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  align-items: ${({ align }) => align};
  justify-content: ${({ justify }) => justify};
`

export default Col
