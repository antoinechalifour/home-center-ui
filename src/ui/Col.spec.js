import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import Col from './Col'

const tests = [
  { align: 'flex-start', justify: 'center' },
  { align: 'flex-end', justify: 'flex-start' },
  { align: 'flex-start', justify: 'space-around' }
]

describe('ui/Col', () => {
  tests.forEach(props =>
    it(`Should render`, () => {
      const tree = renderer.create(
        <Col {...props}>
          <div flex>Hey</div>
          <div>Oh</div>
          <div flex={2}>Lets go!</div>
        </Col>
      )

      expect(tree).toMatchSnapshot()
    })
  )
})
