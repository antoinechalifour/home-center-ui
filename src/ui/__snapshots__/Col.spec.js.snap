// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Col Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

<div
  className="c0"
  direction="column"
>
  <div
    flex="1"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    Hey
  </div>
  <div>
    Oh
  </div>
  <div
    flex="2"
    style={
      Object {
        "flex": 2,
      }
    }
  >
    Lets go!
  </div>
</div>
`;

exports[`ui/Col Should render 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100%;
}

<div
  className="c0"
  direction="column"
>
  <div
    flex="1"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    Hey
  </div>
  <div>
    Oh
  </div>
  <div
    flex="2"
    style={
      Object {
        "flex": 2,
      }
    }
  >
    Lets go!
  </div>
</div>
`;

exports[`ui/Col Should render 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 100%;
}

<div
  className="c0"
  direction="column"
>
  <div
    flex="1"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    Hey
  </div>
  <div>
    Oh
  </div>
  <div
    flex="2"
    style={
      Object {
        "flex": 2,
      }
    }
  >
    Lets go!
  </div>
</div>
`;
